tests/test_bookmarks.py

```diff
@@ -0,0 +1,211 @@
+                            for record in first_sync_messages:
+                                if parent_record == record.get(parent_value):
+                                    records_sync_1.append(record)
+                            for record in second_sync_messages:
+                                if parent_record == record.get(parent_value):
+                                    records_sync_2.append(record)
+                        else:
+                            # as the stream is not child stream, no need to pre-process the records
+                            records_sync_1 = first_sync_messages
+                            records_sync_2 = second_sync_messages
+
+                        second_start_date = new_state['bookmarks'][stream][key]
+
+                        # validate records
+                        self.validate_records(stream, records_sync_1, records_sync_2, value, second_start_date)
+                else:
+                    # Verify the syncs do not set a bookmark for full table streams
+                    self.assertIsNone(first_bookmark_key_value)
+                    self.assertIsNone(second_bookmark_key_value)
+
+                    # Verify the number of records in the second sync is the same as the first
+                    self.assertEqual(second_sync_count, first_sync_count)
+
+    def validate_records(self, stream, records_sync_1, records_sync_2, bookmark, second_start_date):
+        """
+            Validate the records from both syncs with bookmark in the state file
+            and verify we collected records from 1st state bookmark for 2nd sync
+        """
+
+        # collect information specific to incremental streams from syncs 1 & 2
+        replication_key = next(iter(self.expected_replication_keys()[stream]))
+        bookmark_value_parsed = parse(bookmark).strftime('%Y-%m-%dT%H:%M:%SZ')
+
+        # Verify the first sync sets a bookmark of the expected form
+        self.assertIsNotNone(bookmark)
+
+        for record in records_sync_2:
+
+            # Verify the second sync bookmark value is the max replication key value for a given stream
+            replication_key_value = record.get(replication_key)
+            replication_key_value_parsed = parse(replication_key_value).strftime('%Y-%m-%dT%H:%M:%SZ')
+            self.assertLessEqual(
+                replication_key_value_parsed, bookmark_value_parsed,
+                msg='Second sync bookmark was set incorrectly, a record with a greater replication-key value was synced.'
+            )
+
+            # Verify the data of the second sync is greater-equal to the bookmark from the first sync
+            # We have added 'second_start_date' as the bookmark, it is more recent than
+            #   the default start date and it will work as a simulated bookmark
+            self.assertGreaterEqual(
+                replication_key_value_parsed, parse(second_start_date).strftime('%Y-%m-%dT%H:%M:%SZ'),
```

Comment:
Using `start_date` in variables when referring to state values is a bit confusing. That term should be reserved for historical syncs.

Reply:
Updated the variable.